keyboards/0xcb/1337/keymaps/conor/keymap.c

```diff
@@ -0,0 +1,169 @@
+        0,  0, 31, 49, 64, 78, 64, 49, 31,  0,  0, 97, 22,  8, 22, 97,  0,  0, 31, 49, 96, 64, 64, 96, 32,  0,  0,127, 68, 68,100, 59,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 64, 64,127, 64, 64,  0,  0,  0, 32,100, 68, 68,110, 59,  0,  0, 32,100, 68, 68,110, 59,  0,  0,  0,  0,  0,126,  3,  1,  0,  0,  0,  8, 20, 35, 20,  8,  0,  0,  0,  0,  0,  0,  0,  0,  1,  1,  3,  7,  7, 15,254, 30, 28, 28, 28,255, 28, 28, 28, 30,254, 15,  7,  3,  3,  1,  1,  0,  0,  0,  0,  0,  0,  0,  0,  4, 10, 17, 10,  4,  0,
+        0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 16, 40, 68, 40, 16, 16, 16, 16, 16, 16, 31,  0,  0, 16, 40, 71, 40, 16,  0,  0, 31, 16, 16, 16, 16, 16, 16, 40, 68, 40, 16,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+    };
+    oled_write_raw_P(raw_logo, sizeof(raw_logo));
+}
+static void render_logo_font(void) {
+    static const char PROGMEM qmk_logo[] = {
+        0xC0, 0xC1, 0xC2, 0xC3, 0xC4, 0xCB, 0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0xCB, 0xCB, 0xCB, 0x9C, 0x9D, 0xCB, 0xCB,
+        0xE0, 0xE1, 0xE2, 0xE3, 0xE4, 0xCB, 0xA0, 0xA1, 0xA2, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xCB, 0xCB, 0xBB, 0xBC, 0xBD, 0xBE, 0xCB,
+        0xCC, 0xCD, 0xCE, 0xCF, 0xD0, 0xCB, 0x88, 0x89, 0x8A, 0x8B, 0x8A, 0x8B, 0x8C, 0x8D, 0xCB, 0xCB, 0xDB, 0xDC, 0xDD, 0xDE, 0xCB,
+        0xEC, 0xED, 0xEE, 0xEF, 0xF0, 0xCB, 0xA8, 0xA9, 0xAA, 0xAB, 0xAA, 0xAB, 0xAC, 0xAD, 0xCB, 0xFA, 0xFB, 0xFC, 0xFD, 0xFE, 0xFF, 0x00
+    };
+
+    oled_write_P(qmk_logo, false);
+}
+/* Shows the name of the current layer and locks for the host (CAPS etc.) */
+static void render_info(void) {
+    oled_write_P(PSTR("Layer: "), false);
+
+    switch (get_highest_layer(layer_state)) {
+        case _HOME:
+            oled_write_ln_P(PSTR("HOME"), false);
+            break;
+        case _MISC:
+            oled_write_ln_P(PSTR("MISC"), false);
+            break;
+        case _RGB:
+            oled_write_ln_P(PSTR("RGB"), false);
+            break;
+        case _BLED:
+            oled_write_ln_P(PSTR("Backlight"), false);
+            break;
+        default:
+            oled_write_ln_P(PSTR("Undefined"), false);
+    }
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_ln_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+static void render_rgbled_status(bool full) {
+#ifdef RGBLIGHT_ENABLE
+  char buf[30];
+  if (RGBLIGHT_MODES > 1 && rgblight_is_enabled() && get_highest_layer(layer_state) == _RGB) {
+      if (full) {
+          snprintf(buf, sizeof(buf), "RGB mode %2d: %d,%d,%d  \n",
```

Comment:
Just a heads up, snprintf can use 1.5k of firmware space.  Instead, breaking it up into sections and then just rendering numbers like this:
```c
    char string[4];
    uint16_t n = get_value();
    string[3] = '\0';
    string[2] = '0' + n % 10;
    string[1] = ( n /= 10) % 10 ? '0' + (n) % 10 : (n / 10) % 10 ? '0' : ' ';
    string[0] =  n / 10 ? '0' + n / 10 : ' ';
    oled_write(string, false);
```